### PR TITLE
added meson support

### PR DESCRIPTION
**Pull request by noah1510**
This pull request adds full support for use with the meson build system.

All the dependencies are automatically checked and if required even compiled.

It is possible to enable or disable the build for the sample code and the tests.

The library, samples and tests are subprojects.
Since all subprojects are required to be in the subprojects folder there are symbolic links to the actual location of the folder.

For meson the three test sources where split in individual test programs that are called if `meson test` is called.
If one test function fails, it will be reported that 2 out of 3 test were successful.

The readme was updated with instructions for using meson and the syntax problem in it where fixed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,7 +53,58 @@ sudo apt update && sudo apt upgrade
...
 
-### Build
+------
+
+## Build
+
+There are two ways to compile eitehr with CMake or with meson.
+By default meson is faster, but if your project uses CMake, those instructions might be better for integration.
+
+### Building with meson
```
Comment (h4tr3d) [inline README.md:63]: Could you please move Meson section after CMake section? CMake support is prioritized. At least for now due to better IDE integration than Meson projects.

Comment (noah1510) [reply README.md:103]: Fixed in 6b1a7eb

**Comment by h4tr3d** [comment]
Make a signal, when work will be completed. I am ready to integrate these changes.

**Comment by noah1510** [comment]
I will add it to the Travis build after that everything should be done and work just fine

**Comment by h4tr3d** [comment]
> I will add it to the Travis build after that everything should be done and work just fine

Ok, thanks. Just wait for work completion.

**Comment by noah1510** [comment]
travis ci should be implemented.
Could you look over it since I am not very familiar with travis, since I usually use github actions?

**Comment by h4tr3d** [comment]
>  Could you look over it since I am not very familiar with travis, since I usually use github actions?

Yes. I will do it a little after. Can I merge it to main tree now?

**Comment by noah1510** [comment]
If the travis ci looks fine you can go ahead and merge it

**Pull request merged**